airwallex/src/main/java/com/airwallex/android/view/PaymentMethodsViewModel.kt

```diff
@@ -157,43 +171,30 @@ internal class PaymentMethodsViewModel(
 
     suspend fun fetchAvailablePaymentMethodsAndConsents():
             Result<Pair<List<AvailablePaymentMethodType>, List<PaymentConsent>>>? {
-        return when (session) {
-            is AirwallexPaymentSession, is AirwallexRecurringWithIntentSession -> {
-                paymentIntent?.clientSecret
+        val secret = clientSecret.takeIf { !it.isNullOrBlank() }
+            ?: return Result.failure(AirwallexCheckoutException(message = "Client secret is empty or blank"))
```

Comment:
Shall we even bring this client secret check forward since it will fail anyway? i.e. Make `clientSecret` field in `PaymentIntent` mandatory.